src/main/java/com/minecolonies/core/datalistener/DiseasesListener.java

```diff
@@ -0,0 +1,234 @@
+import io.netty.buffer.Unpooled;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.server.packs.resources.ResourceManager;
+import net.minecraft.server.packs.resources.SimpleJsonResourceReloadListener;
+import net.minecraft.util.GsonHelper;
+import net.minecraft.util.RandomSource;
+import net.minecraft.util.profiling.ProfilerFiller;
+import net.minecraft.world.item.ItemStack;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+
+/**
+ * Loads and listens to diseases data.
+ */
+public class DiseasesListener extends SimpleJsonResourceReloadListener
+{
+    /**
+     * Gson instance
+     */
+    private static final Gson GSON = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
+
+    /**
+     * Json constants
+     */
+    private static final String KEY_NAME   = "name";
+    private static final String KEY_RARITY = "rarity";
+    private static final String KEY_ITEMS  = "items";
+
+    /**
+     * The map of diseases.
+     */
+    private static RandomCollection<ResourceLocation, Disease> DISEASES = new RandomCollection<>();
+
+    /**
+     * A possible disease.
+     *
+     * @param id        the id of the disease.
+     * @param name      the name of the disease.
+     * @param rarity    the rarity of the disease.
+     * @param cureItems the list of items needed to heal.
+     */
+    public record Disease(ResourceLocation id, Component name, int rarity, List<ItemStorage> cureItems)
+    {
+        /**
+         * Get the cure string containing all items required for the cure.
+         *
+         * @return the cure string.
+         */
+        public Component getCureString()
+        {
+            final MutableComponent cureString = Component.literal("");
+            for (int i = 0; i < cureItems.size(); i++)
+            {
+                final ItemStorage cureStack = cureItems.get(i);
+                cureString.append(String.valueOf(cureStack.getItemStack().getCount())).append(" ").append(cureStack.getItemStack().getHoverName());
+                if (i != cureItems.size() - 1)
+                {
+                    cureString.append(" + ");
+                }
+            }
+            return cureString;
+        }
+    }
+
+    /**
+     * Default constructor.
+     */
+    public DiseasesListener()
+    {
+        super(GSON, "colony/diseases");
+    }
+
+    /**
+     * Sync to client.
+     *
+     * @param player to send it to.
+     */
+    public static void sendGlobalDiseasesPackets(final ServerPlayer player)
+    {
+        final FriendlyByteBuf byteBuf = new FriendlyByteBuf(Unpooled.buffer());
```

Comment:
why a buffer here? Serialization should normally happen in the network thread within the packet, and you pass the packet the needed information to serialize

Reply:
I'm fine with either, but then I'm changing all datalisteners